lib/assessment.js

```diff
@@ -29,7 +46,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  var ary2 = [];
+  for (var i = 0; i < ary2.length; i++){
+    ary2[] = defaultsFunction(i);
+  }
+  return ary2[];
```

Comment:
Also, this should be `return ary2`, since `ary2` is the name of the variable that's holding the array.